src/lib.rs

```diff
@@ -61,12 +72,18 @@ impl BiosBoot {
     fn create_fat_partition(&self) -> anyhow::Result<NamedTempFile> {
         let stage_3_path = Path::new(env!("BIOS_STAGE_3_PATH"));
         let stage_4_path = Path::new(env!("BIOS_STAGE_4_PATH"));
+        let has_rd_path = self.ramdisk.is_some();
+        let binding = self.ramdisk.as_deref();
+        let ramdisk_path = binding.unwrap_or(Path::new("no-such-file"));
+        let kernel_path = self.kernel.as_path();
 
         let mut files = BTreeMap::new();
-        files.insert(KERNEL_FILE_NAME, self.kernel.as_path());
+        files.insert(KERNEL_FILE_NAME, kernel_path);
         files.insert(BIOS_STAGE_3, stage_3_path);
         files.insert(BIOS_STAGE_4, stage_4_path);
-
+        if has_rd_path {
+            files.insert(RAMDISK_FILE_NAME, ramdisk_path);
+        }
```

Comment:
This seems like a good use case for `if let`:

```suggestion
        if let Some(ramdisk_path) = &self.ramdisk {
            files.insert(RAMDISK_FILE_NAME, ramdisk_path);
        }
```

This way, we don't need the additional bindings in lines 75-78.